community/cypher/cypher-compiler-3.3/src/main/scala/org/neo4j/cypher/internal/compiler/v3_3/planner/logical/plans/LogicalPlan.scala

```diff
@@ -45,6 +45,24 @@ abstract class LogicalPlan
   def solved: PlannerQuery with CardinalityEstimation
   def availableSymbols: Set[IdName]
 
+  def assignedId: LogicalPlanId = _id.getOrElse(throw new InternalException("Plan has not had an id assigned yet"))
+  def assignIds(): Unit = recurseAssignIds(new LogicalPlanId(0))
+
+  private var _id: Option[LogicalPlanId] = None
+  protected def assignId(id: Option[LogicalPlanId]): Unit = _id = id
+
+  protected def recurseAssignIds(sofar: LogicalPlanId): LogicalPlanId = {
```

Comment:
Aren't we running into the stack-depth problem if recursing over a huge logical plan here? Not saying that we have too fix it here and now, it might be better to get this operator in asap.